hphp/runtime/vm/jit/vasm-simplify-arm.cpp

```diff
@@ -99,6 +99,25 @@ bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool simplify(Env& env, const ldimmq& inst, Vlabel b, size_t i) {
+  return if_inst<Vinstr::lea>(env, b, i + 1, [&] (const lea& ea) {
+    // ldimmq{s, tmp}; lea{tmp, d} -> subqi{s, d}
+    if (!(env.use_counts[inst.d] == 1 &&
+          inst.s.q() <= -1  &&
+          inst.s.q() >= -4095 &&
+          ea.s.disp == 0 &&
+          ea.s.base.isValid())) return false;
+
+    return simplify_impl(env, b, i, [&] (Vout& v) {
+      auto sf = env.unit.makeReg();
+      v << subqi{-inst.s.l(), ea.s.base, ea.d, sf};
+      return 2;
+    });
```

Comment:
What is this simplification intended to do?  Currently, it could transform the following sequence:

```
ldimmq{-42, rimm};
lea{rvmfp()[0], d};
store{rimm, m};
```

into:

```
subqi{42, rvmfp(), d, sf};
store{rimm, m};
```

This is totally nonsensical and unsound, and I'm not really sure what you meant to do here.